password_security/models/res_users.py

```diff
@@ -72,6 +76,31 @@ def password_match_message(self):
     def _check_password(self, password):
         self._check_password_rules(password)
         self._check_password_history(password)
+        self._check_password_similar_name(password, self.name)
+        self._check_password_similar_login(password, self.login)
+        return True
+
+    @api.multi
+    def _check_password_similar_name(self, password, name):
+        self.ensure_one()
+        if self.company_id.password_not_similar_name_login:
+            # remove non-alnums from user's name for comparison
+            name_alnum = re.sub(r'[\W_]', '', name)
+            if name_alnum.lower() == password.lower():
+                raise PassError(self.password_match_message())
+        return True
+
+    @api.multi
+    def _check_password_similar_login(self, password, login):
+        self.ensure_one()
+        if self.company_id.password_not_similar_name_login:
+            match = re.match(r'([^@]+)(?:@[^@]+)?', login)
+            login_name_only = match.group(1)
+            login_whole = match.group(0)
+            if login_name_only.lower() == password.lower():
```

Comment:
But that is checking equality, not similarity :thinking:

Reply:
OK, case- and non-alnums-ignoring equality... Maybe some common subsequence or substring matching would do better.

Reply:
check https://docs.python.org/3/library/re.html

Reply:
For LCS i'd prefer [difflib.SequenceMatcher](https://docs.python.org/3/library/difflib.html#difflib.SequenceMatcher) . Or did you mean that I'm doing something wrong with [re](https://docs.python.org/3/library/re.html)?

Reply:
nothing wrong

Reply:
I guess equality is a good 1st step towards similarity.